server/controllers/finance/exchange.js

```diff
@@ -22,11 +25,47 @@ function getExchangeRate(enterpriseId, currencyId, date) {
     .then(rows => rows[0]);
 }
 
-// gets a positive number for the exchange rate display.
+// gets a non-fractional number for the exchange rate display.
 function formatExchangeRateForDisplay(value) {
   return (value < 1) ? util.roundDecimal(1 / value, 2) : value;
 }
 
+// Construct the HTML string to show the exchange rate in a nice format
+function exchangeRateMsg(currencyId, rate, enterprise, lang, date) {
```

Comment:
I think this approach is close, but not optimal in my view.  Would it be possible to make this into a [handlebars helper](https://handlebarsjs.com/guide/partials.html#basic-partials)?  That seems like a better approach.  

Here is why:

1. Within the handlebars templates, you already have access to certain metadata - the enterprise (and therefore the currency), the language, etc.  You would need to add target currency and date.
2. You can use the handlebars translate + currency rendering to get both those elements for free.
3. We have less HTML embedded in JS, which makes it hard to lint, read, and debug.
3. It should be much easier to integrate this into other reports.

I would propose that you break this out to a handlebars partial.  If you need to attach a common object to all reports (like we do already in `ReportManager` with the `metadata` tag), please do it.

https://github.com/IMA-WorldHealth/bhima/blob/f35a8dfdf0556bda0bc35da9e805c58641e45ce2/server/lib/ReportManager.js#L133-L135

Reply:
Concretely, rather than adding:
```handlebars
{{{ exchangeRateMsg }}}
```

It would be better to have:
```handlebars
{{#exhangeRateMessage foreignCurrency=currency rate=rate}}
```

Reply:
👍 I'll work on it.  Thanks for the suggestion.   I was not sure how to do it as a handlebars helper...

Reply:
Done